### PR TITLE
435 multiple shopping lists

### PR DESCRIPTION
**Pull request by lythix**
Have added two new buttons (Save and Import) to the Shopping List panel, and a new button on the Import/Export for importing a shopping list previously saved. 

Save button will prompt for filename and save the blueprints that make up the current shopping list. Import button will ask user if they want to clear current shopping list, or import blueprints without clearing. Seems to work quite nicely, but happy for any feedback from additional testing.

### CODE REVIEW

**Comment by msarilar** [comment]
excellent, thanks for that !

I'll review this thorougly (I've been away for a while and should come back in around a week)

**Comment by lythix** [comment]
> excellent, thanks for that !
> 
> I'll review this thorougly (I've been away for a while and should come back in around a week)

It's been a long week. Any issues with this (I see there are merge conflicts now)

**Comment by msarilar** [comment]
hey sorry for the long wait, I had a look back then but didn't get back to it

I wonder if maybe it would be better to use the app's local directory (see `GetManualChangesDirectory`) instead of having the user choose his own, wdyt? that way it stays self contained and EDEngineer doesn't have files/config hanging around

**Comment by codan-prices** [comment]
> hey sorry for the long wait, I had a look back then but didn't get back to it

All good, we get busy. I forgot about it too. :)

> 
> I wonder if maybe it would be better to use the app's local directory (see `GetManualChangesDirectory`) instead of having the user choose his own, wdyt? that way it stays self contained and EDEngineer doesn't have files/config hanging around

Sure, sounds good. I'll have a look at making that the default directory and fix up the merge conflicts while I'm at it. Thanks!

**Comment by msarilar** [comment]
I fixed the conflicts earlier in the branch

**Comment by lythix** [comment]
> I fixed the conflicts earlier in the branch

Have changed the default directory. I've got a new laptop since I last built this branch, and am having some issues with some missing dependencies so can't test it right now. 

I seem to be missing Microsoft.WindowsAPICodePack and possibly some other packages. I'll see if I can work out how I got it building before.

File: EDEngineer/Views/MainWindow.xaml [hunk 1]
```diff
@@ -198,7 +198,7 @@
                                          Background="Transparent"
                                          VerticalAlignment="Center"
                                          BorderBrush="Transparent"
-                                         FontSize="15"
+                                         FontSize="35"
```
[the remark this answers is not in the document]
Comment (lythix) [reply EDEngineer/Views/MainWindow.xaml:201]:
```suggestion
                                         FontSize="15"

```
Not one that I remember. Happy to revert

**Comment by lythix** [comment]
Have addressed feedback. 

I'm still having issues building on two different machines. It's been a while since I've done any WPF. 
Not found a fix yet but wondering if you can tell me where your Windows reference path leads to. 
image
As you can see in screenshot, my reference for that dll has an issue finding it which I suspect is my issue building. 

Results in the Windows.UI.Notification using statement not being able to be found
image

**Comment by msarilar** [comment]
thanks very much!

for the Windows.UI.Notifications I think you have to install some UWP SDK or something

**Pull request merged**